Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,190 @@
+
+        /// <summary>
+        /// Gets or sets the DeviceList Selection Mode. Defaults to ListViewSelectionMode.Single
+        /// </summary>
+        public ListViewSelectionMode SelectionMode
+        {
+            get { return (ListViewSelectionMode)GetValue(SelectionModeProperty); }
+            set { SetValue(SelectionModeProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="SelectionMode"/>.
+        /// </summary>
+        public static readonly DependencyProperty SelectionModeProperty = DependencyProperty.Register(
+            nameof(SelectionMode),
+            typeof(ListViewSelectionMode),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(ListViewSelectionMode.Single));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class.
+        /// </summary>
+        public RemoteDevicePicker()
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            SecondaryButtonText = "Done";
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
+            DeviceMap = new Dictionary<string, RemoteSystem>();
+        }
+
+        /// <summary>
+        /// Initiate Picker UI
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<List<RemoteSystem>> PickDeviceAsync()
+        {
+            List<RemoteSystem> remoteSystems = new List<RemoteSystem>();
+            await ShowAsync();
+            foreach (RemoteSystem system in listDevices.SelectedItems)
+            {
+                remoteSystems.Add(system);
+            }
+
+            return remoteSystems;
+        }
+
+        /// <inheritdoc />
+        protected async override void OnApplyTemplate()
+        {
+            listDevices = GetTemplateChild("PART_ListDevices") as ListView;
+            listDeviceTypes = GetTemplateChild("PART_ListDeviceTypes") as ComboBox;
+            progressRing = GetTemplateChild("PART_ProgressRing") as ProgressRing;
+
+            var enumval = Enum.GetValues(typeof(DeviceType)).Cast<DeviceType>();
+            listDeviceTypes.ItemsSource = enumval.ToList();
+            listDeviceTypes.SelectionChanged += ListDeviceTypes_SelectionChanged;
+            listDeviceTypes.SelectedIndex = 0;
+
+            RemoteSystemAccessStatus accessStatus = await RemoteSystem.RequestAccessAsync();
+            if (accessStatus == RemoteSystemAccessStatus.Allowed)
+            {
+                RemoteSystemWatcher m_remoteSystemWatcher = RemoteSystem.CreateWatcher();
+                m_remoteSystemWatcher.RemoteSystemAdded += RemoteSystemWatcher_RemoteSystemAdded;
+                m_remoteSystemWatcher.RemoteSystemRemoved += RemoteSystemWatcher_RemoteSystemRemoved;
+                m_remoteSystemWatcher.RemoteSystemUpdated += RemoteSystemWatcher_RemoteSystemUpdated;
```

Comment:
Yes. Instead of `EnumerationCompleted` Event, I am using `RemoteSystemsUpdated` Event to make the ProgressRing IsActive property to False. I explicitly chose to use this Event so that I can show the progress ring until the ListView is updated. Let me know if it is not a good idea.

Reply:
RemoteSystemUpdated is going to be fired when there is an update to existing / already-discovered `RemoteSystem`, But my question here is how are you going to detect that the discovery has finished  without using `EnumerationCompleted` ? Also I haven't found any code in the control that stop the discovery, which we totally should do.